h2o-core/src/main/java/water/rapids/ast/prims/filters/dropduplicates/DropDuplicateRows.java

```diff
@@ -0,0 +1,89 @@
+package water.rapids.ast.prims.filters.dropduplicates;
+
+import water.DKV;
+import water.Key;
+import water.Scope;
+import water.fvec.Frame;
+import water.fvec.Vec;
+import water.rapids.Merge;
+import water.util.ArrayUtils;
+import water.util.FrameUtils;
+
+import java.util.Arrays;
+
+/**
+ * Drops duplicated rows of a Frame
+ */
+public class DropDuplicateRows {
+
+  private static final String LABEL_COLUMN_NAME = "label";
+
+  final Frame sourceFrame;
+  final int[] comparedColumnIndices;
+  final KeepOrder keepOrder;
+
+  /**
+   * @param sourceFrame           Frame to perform the row de-duplication on
+   * @param comparedColumnIndices Indices of columns to consider during the comparison
+   * @param keepOrder             Which rows to keep.
+   */
+  public DropDuplicateRows(final Frame sourceFrame, final int[] comparedColumnIndices, final KeepOrder keepOrder) {
+    this.sourceFrame = sourceFrame;
+    this.comparedColumnIndices = comparedColumnIndices;
+    this.keepOrder = keepOrder;
+  }
+
+  public Frame dropDuplicates() {
+    Frame outputFrame = null;
+    try {
+      Scope.enter();
+      FrameUtils.labelRows(sourceFrame, LABEL_COLUMN_NAME);
+      final Frame sortedFrame = Scope.track(sortByComparedColumns());
+      final Frame chunkBoundaries = Scope.track(new CollectChunkBorderValuesTask()
+              .doAll(sortedFrame.types(), sortedFrame)
+              .outputFrame());
+
+      final Frame deDuplicatedFrame = Scope.track(new DropDuplicateRowsTask(chunkBoundaries, comparedColumnIndices)
+              .doAll(sortedFrame.types(), sortedFrame)
+              .outputFrame(Key.make(), sourceFrame.names(), sourceFrame.domains())); // Removing duplicates, domains remain the same
+
+      outputFrame = Merge.sort(deDuplicatedFrame, deDuplicatedFrame.numCols() - 1);
+      outputFrame.remove(outputFrame.numCols() - 1);
+      outputFrame._key = sourceFrame._key != null ? Key.make(sourceFrame._key.toString() + "_deduplicated") : Key.make();
```

Comment:
imho the output frame doesn't need a key and shouldn't be in DKV - rapids will take of that

Reply:
No it doesn't. Removed, also removed testing presence of a Key from JUnit tests.